prover/maths/fft/fft.go

```diff
@@ -14,19 +16,26 @@ const (
 	DIF
 )
 
```

Comment:
What are the diffs between this version of the fft and gnark's ? It may be interesting to merge the two I believe

Reply:
Presented improvements mostly migrates all logic from gnrak-crypto, except of some tricks with twiddles that will require changes in the twiddles generation that may not be suitable for your in-place implementation (imho).